### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by dhardy**
Just a copy from https://github.com/kas-gui/blog so should be fine.

### CODE REVIEW

**Comment by dhardy** [comment]
This runner only does anything on the `master` branch, hence no jobs.

I could add one for `mdbook test`, but there are a *lot* of issues we'd need to fix first (broken examples and relative links).

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: '0.4.4'
+          # mdbook-version: 'latest'
```
Comment (newpavlov) [inline .github/workflows/gh-pages.yml:18]: Is there a reason why `mdbook-version` is not set to `latest`?

Comment (dhardy) [reply .github/workflows/gh-pages.yml:18]: Because builds might break if there are breaking changes?

**Comment by dhardy** [comment]
@newpavlov can you approve please?

**Review by newpavlov — APPROVED** [review]

**Pull request merged**